hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -146,11 +146,15 @@ public CleanPlanner(HoodieEngineContext context, HoodieTable<T, I, K, O> hoodieT
     if (config.incrementalCleanerModeEnabled()) {
       Option<HoodieInstant> lastClean = hoodieTable.getCleanTimeline().filterCompletedInstants().lastInstant();
       if (lastClean.isPresent()) {
-        HoodieCleanMetadata cleanMetadata = TimelineMetadataUtils
-            .deserializeHoodieCleanMetadata(hoodieTable.getActiveTimeline().getInstantDetails(lastClean.get()).get());
-        if ((cleanMetadata.getEarliestCommitToRetain() != null)
-            && (cleanMetadata.getEarliestCommitToRetain().length() > 0)) {
-          return getPartitionPathsForIncrementalCleaning(cleanMetadata, instantToRetain);
+        if (hoodieTable.getActiveTimeline().isEmpty(lastClean.get())) {
+          hoodieTable.getActiveTimeline().deleteEmptyIfExists(lastClean.get());
```

Comment:
I am wondering how come we could have a empty file for a completed clean. I thought we might have empty file issues only with inflght ones. can you help me understand.

Reply:
Hello, I haven't figured out the root cause yet